src/pages/index.js

```diff
@@ -62,36 +104,114 @@ function handleImageClick(cardData) {
 
 //edit profile
 function handleProfileSubmit(data) {
-  userInfo.setUserInfo({name: data.name, job: data.description});
-editPopup.close(); }
-
-//add card
+  console.log(data);
+  api.updateUserInfo({
+    name: data.name,
+    about: data.about
+  })
+    .then(updatedUserData => {
+      user.setUserInfo({
+        name: updatedUserData.name, 
+        about: updatedUserData.about || updatedUserData.dascription || updatedUserData.bio
+      });
+      editPopup.close();
+    })
+    .catch(err => console.error(`Error updating profile: ${err}`));
+   }
+
+// Add card
 function handleAddCardSubmit(data) {
-  const card = {
+  const cardData = {
     name: data.title,
     link: data.url
   };
-  const cardElement = createCard(card);
-  section.addItems(cardElement);
-  addcardPopup.close();
-  profileAddFormElement.reset();
-  addFormValidator.disableButton();
+
+  // Create the Card 
+  api.addNewCard(cardData)
+    .then((newCardData) => {
+      const cardElement = createCard(newCardData);
+      section.addItems(cardElement);  
+      addCardPopup.close();            
+      profileAddFormElement.reset();  
+      addFormValidator.disableButton();  
+    })
+    .catch(err => {
+      console.error(`Error adding card: ${err}`);
+    });
+}
+
+
+
+//avatar
+function handleAvatarSubmit(data){
+  api.updateUserAvatar({
+    avatar: data.url
+  })
+    .then(updateUserData => {
+      user.setUserInfo({
+        avatar: updateUserData.avatar
+      });
+      avatarPopup.close();
+    })
+    .catch(err => console.error(`Error updating avatar: ${err}`));
 }
 
-function openEditProfileModal() {
-  const currentUser = userInfo.getUserInfo();
+//delete
+function handleDeleteSubmit(card){
+  const confirmButton = document.querySelector('#delete-confirm-button');
+  deletePopup.open();
+  confirmButton.addEventListener('click', function onConfirmClick() {
+    card.removeCard();
+    api.deleteCard(card._id)
+      .then(card => {
+          deletePopup.close();
+          confirmButton.removeEventListener('click', onConfirmClick)
+      })
+      .catch(err => {
+        console.error(`Error deleting card ${err}`);
+    });
+  })
+}
+  
+    
+    
+  
+
+//like
+function handleCardLike(card){
+  const isLiked = card.isLiked(); 
+
+  const likePromise = isLiked
+    ? api.removeLike(card.getId())  
+    : api.addLike(card.getId()); 
+
+  likePromise
+    .then(updatedCardData => {
+      card.updateLikes(updatedCardData.isLiked);  
+    })
+    .catch(err => console.error(`Error updating like status: ${err}`));
+}
+
+function openModal(popup) {
+  const currentUser = user.getUserInfo();
   userNameInput.value = currentUser.name;
-  userJobInput.value = currentUser.job;
-  editPopup.open();
+  userJobInput.value = currentUser.about;
+  popup.open();
 }
 
 // Event Listeners
-profileEditButton.addEventListener('click', openEditProfileModal);
+avatar.addEventListener('click', () =>{openModal(avatarPopup)})
```

Comment:
`openModal` has the profile logics, so you can't use it for the avatar


Please, add `avatarPopup.open();` here


```js

avatar.addEventListener("click", () => {
  avatarPopup.open();
});

```